pkg/proxy/handlers/spacelister.go

```diff
@@ -265,3 +264,43 @@ func errorResponse(ctx echo.Context, err *apierrors.StatusError) error {
 	ctx.Response().Writer.WriteHeader(int(err.ErrStatus.Code))
 	return json.NewEncoder(ctx.Response().Writer).Encode(err.ErrStatus)
 }
+
+// filterUserSpaceBindings returns all the spacebindings for a given username
+func filterUserSpaceBindings(username string, allSpaceBindings []toolchainv1alpha1.SpaceBinding) (outputBindings []toolchainv1alpha1.SpaceBinding) {
+	for _, binding := range allSpaceBindings {
+		if binding.Spec.MasterUserRecord == username {
+			outputBindings = append(outputBindings, binding)
+		}
+	}
+	return outputBindings
+}
+
+// generateWorkspaceBindings generates the bindings list starting from the spacebindings found on a given space resource and an all parent spaces.
+// The Bindings list  has the available actions for each entry in the list.
+func generateWorkspaceBindings(space *toolchainv1alpha1.Space, spaceBindings []toolchainv1alpha1.SpaceBinding) []toolchainv1alpha1.Binding {
+	var bindings []toolchainv1alpha1.Binding
+	for _, spaceBinding := range spaceBindings {
+		binding := toolchainv1alpha1.Binding{
+			MasterUserRecord: spaceBinding.Spec.MasterUserRecord,
+			Role:             spaceBinding.Spec.SpaceRole,
+			AvailableActions: []string{},
+		}
+		if name, exists := spaceBinding.GetLabels()[toolchainv1alpha1.SpaceBindingSpaceLabelKey]; exists && name == space.GetName() {
+			// this is a binding that was created on the current space, it can be updated or deleted
+			binding.AvailableActions = []string{UpdateBindingAction, DeleteBindingAction}
```

Comment:
how does the client (UI) know which SBR it should either update or delete?
I think that the API is missing the `bindingRequest` field, right?
https://docs.google.com/document/d/1hWaFvCfsEXrbqs8ndDR0JPjKROXYJkvj1ZCnio4P1WU/edit#bookmark=id.tttpu2pbe4a6

Reply:
Ohh jeez 🤦‍♂️ , I missed that field in the original API pr and totally forgot about it! Great catch! 

I've raised https://github.com/codeready-toolchain/api/pull/378 for this. And I'm updating this PR and e2e tests to use it! 

Thanks!

Reply:
I'm now populating the `bindingRequest` field as well. PTAL at https://github.com/codeready-toolchain/registration-service/pull/350/commits/64ba7bab696d459bc4b170749685a926b15b6f0e